{"success": true, "error": null, "issues": [{"title": "Ether send", "description": "In the function 'collectPercentOfFees(uint256)' a non-zero amount of Ether is sent to an address taken from storage slot 5There is a check on storage index 5. This storage slot can be written to by calling the function 'DynamicPyramid()'.\n\nThere is a check on storage index 6. This storage slot can be written to by calling the function 'DynamicPyramid()'.\nThere is a check on storage index 7. This storage slot can be written to by calling the function 'fallback'.", "function": "collectPercentOfFees(uint256)", "type": "Warning", "address": 1599, "debug": "SOLVER OUTPUT:\ncalldata_Rubixi_4: 0x0\nstorage_1: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\nstorage_5: 0x0\ncaller: 0x0\ncalldata_Rubixi_0: 0x4229616d00000000000000000000000000000000000000000000000000000000\ncalldatasize_Rubixi: 0x4\ncallvalue: 0x0\n", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 93, "code": "creator.send(feesToCollect)"}, {"title": "Ether send", "description": "In the function 'collectAllFees()' a non-zero amount of Ether is sent to an address taken from storage slot 5There is a check on storage index 5. This storage slot can be written to by calling the function 'DynamicPyramid()'.\n\nThere is a check on storage index 9. This storage slot can be written to by calling the function 'DynamicPyramid()'.\nThere is a check on storage index 10. This storage slot can be written to by calling the function 'fallback'.", "function": "collectAllFees()", "type": "Warning", "address": 1940, "debug": "SOLVER OUTPUT:\nstorage_1: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\nstorage_5: 0x0\ncaller: 0x0\ncalldata_Rubixi_0: 0x686f2c9000000000000000000000000000000000000000000000000000000000\ncalldatasize_Rubixi: 0x4\ncallvalue: 0x0\n", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 75, "code": "creator.send(collectedFees)"}, {"title": "Exception state", "description": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that assert() should only be used to check invariants. Use require() for regular input checking. ", "function": "nextPayoutWhenPyramidBalanceTotalsApproximately()", "type": "Informational", "address": 1653, "debug": "The exception is triggered under the following conditions:\n\nstorage_4: 0x0\nstorage_6: 0x0\ncalldata_Rubixi_0: 0x57d4021b00000000000000000000000000000000000000000000000000000000\ncalldatasize_Rubixi: 0x4\ncallvalue: 0x0\n", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 131, "code": "participants[payoutOrder]"}, {"title": "Exception state", "description": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that assert() should only be used to check invariants. Use require() for regular input checking. ", "function": "participantDetails(uint256)", "type": "Informational", "address": 2085, "debug": "The exception is triggered under the following conditions:\n\ncalldata_Rubixi_4: 0x0\nstorage_6: 0x0\ncalldata_Rubixi_0: 0x9dbc4f9b00000000000000000000000000000000000000000000000000000000\ncalldatasize_Rubixi: 0x4\ncallvalue: 0x0\n", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 148, "code": "participants[orderInPyramid]"}, {"title": "Integer Underflow", "description": "A possible integer underflow exists in the function numberOfParticipantsWaitingForPayout().\nThe substraction may result in a value < 0.", "function": "numberOfParticipantsWaitingForPayout()", "type": "Warning", "address": 2743, "debug": "storage_6: 0x0\nstorage_4: 0x1\ncalldata_Rubixi_0: 0xd11f13df00000000000000000000000000000000000000000000000000000000\ncalldatasize_Rubixi: 0x4\ncallvalue: 0x0\n", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 143, "code": "participants.length - payoutOrder"}, {"title": "Unchecked CALL return value", "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.", "function": "collectPercentOfFees(uint256)", "type": "Informational", "address": 1599, "debug": "", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 93, "code": "creator.send(feesToCollect)"}, {"title": "Unchecked CALL return value", "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.", "function": "collectFeesInEther(uint256)", "type": "Informational", "address": 1940, "debug": "", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 75, "code": "creator.send(collectedFees)"}, {"title": "Unchecked CALL return value", "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.", "function": "collectFeesInEther(uint256)", "type": "Informational", "address": 2582, "debug": "", "filename": "<TEST_FILES>/rubixi.sol", "lineno": 85, "code": "creator.send(_amt)"}]}